daemon/endpoint.go

```diff
@@ -233,32 +233,34 @@ func (d *Daemon) createEndpoint(ctx context.Context, epTemplate *models.Endpoint
 		}
 	}
 
+	logger := ep.Logger(daemonSubsys)
 	if err := endpointmanager.AddEndpoint(d, ep, "Create endpoint from API PUT"); err != nil {
-		log.WithError(err).Warn("Aborting endpoint join")
+		logger.WithError(err).Warn("Aborting endpoint join")
 		return nil, err
 	}
 
```

Comment:
You want to call `ep.Logger()` again here so the log messages will have the endpoint ID populated. It is assigned in `endpointmanager.AddEndpoint()`

Reply:
good one, fixed ;-)